ph5/core/tests/test_base.py

```diff
@@ -12,18 +16,57 @@ def initialize_ex(nickname, path, editmode=False):
     return ex
 
 
+class LogTestCase(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        # enable propagating to higher loggers
+        logger.propagate = 1
+        cls.handlers = [h for h in logger.handlers]
+        # switch handler that send log to console
+        # to StringIO handler to catch log's messages in test
+        log = StringIO()
+        new_handler = logging.StreamHandler(log)
+        logger.handlers = [new_handler]
+
+    @classmethod
+    def tearDownClass(cls):
+        # disable propagating to higher loggers
+        logger.propagate = 0
+        # put back the stream handler in ph5.__init__.logger
+        logger.handlers = cls.handlers
+
+    def tearDown(self):
+        # clean up handlers in any ph5 files' loggers that have been used
+        # but exclude logger from ph5/__init__.py to catch log's messages in
+        # next test
+        for k, v in logging.Logger.manager.loggerDict.items():
+            if (not isinstance(v, logging.PlaceHolder)) and ('ph5.' in k):
+                v.handlers = []
+        super(LogTestCase, self).tearDown()
+
+
 class TempDirTestCase(unittest.TestCase):
+    total_error_failure = 0
 
     def setUp(self):
         """
         create tmpdir
         """
+        self.tmpdir = None
+        self.home = None
```

Comment:
These are not needed.
This was for printing these variables before the first run.